pkg/clusteragent/externalmetrics/autoscaler_watcher_test.go

```diff
@@ -60,6 +66,18 @@ func (f *autoscalerFixture) newAutoscalerWatcher() (*AutoscalerWatcher, kube_inf
 		f.kubeObjects = append(f.kubeObjects, hpa)
 	}
 	kubeClient := kube_fake.NewSimpleClientset(f.kubeObjects...)
+	kubeClient.Resources = []*metav1.APIResourceList{
```

Comment:
Can we extend the test suite to cover all the supported api versions?

Reply:
Not sure if unit tests for that would bring a lot of value. I think e2e tests against the different k8s versions are the only way to actually test that support properly.

Reply:
Yes good point. Let's extend `test/e2e/argo-workflows/templates/nginx.yaml` to run multiple tests with different api versions! I know the freeze is approaching so I'll let you decide whether to do it in this PR or later.

Reply:
I'll create a card to address this not under time pressure :)